Libraries/Lists/ListView/ListView.js

```diff
@@ -347,16 +347,23 @@ const ListView = createReactClass({
       contentLength: null,
       offset: 0,
     };
+
     this._childFrames = [];
     this._visibleRows = {};
     this._prevRenderedRowsCount = 0;
     this._sentEndForContentLength = null;
   },
 
+  componentWillUnmount: function() {
+    if (this._rafId != null) {
+      cancelAnimationFrame(this._rafId);
+    }
+  },
+
   componentDidMount: function() {
     // do this in animation frame until componentDidMount actually runs after
     // the component is laid out
-    this.requestAnimationFrame(() => {
+    this._rafId = requestAnimationFrame(() => {
```

Comment:
Both `componentDidMount` and `componentDidUpdate` will write to `this._rafId`, which means that the `rafId` from `componentDidUpdate` will overwrite the `rafId` from `componentDidMount`. Suppose that both hooks are executed, and then the component is unmounted, all before the first `requestAnimationFrame` hook is executed. Then, we'll end up executing `this._measureAndUpdateScrollProps();`. 

To get around this, we could add the following function to ListView:

```JavaScript
requestAnimationFrame(fn: () => void): void {
  const rafId = requestAnimationFrame(() => {
    clearAnimationFrame(rafId);
    this._rafIds.splice(this._rafIds.indexOf(rafId));
    fn();
  })
  this._rafIds.push(rafId);
},
```